helix-rest/src/main/java/org/apache/helix/rest/server/resources/lock/LockAccessor.java

```diff
@@ -0,0 +1,228 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.core.Response;
+
+import com.codahale.metrics.annotation.ResponseMetered;
+import com.codahale.metrics.annotation.Timed;
+import com.google.common.collect.Lists;
+import org.apache.helix.HelixException;
+import org.apache.helix.lock.LockInfo;
+import org.apache.helix.lock.LockScope;
+import org.apache.helix.lock.helix.HelixLockScope;
+import org.apache.helix.lock.helix.ZKDistributedNonblockingLock;
+import org.apache.helix.rest.common.HttpConstants;
+import org.apache.helix.rest.server.resources.helix.AbstractHelixResource;
+import org.apache.helix.zookeeper.datamodel.ZNRecord;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/locks")
+public class LockAccessor extends AbstractHelixResource {
+  private static final Logger LOG = LoggerFactory.getLogger(LockAccessor.class);
+  private static final String ZK_ADDR = "zkAddress";
+  private static final String LOCK_PATH = "lockPath";
+  private static final String LOCK_TIMEOUT = "lockTimeout";
+  private static final String USER_ID = "userId";
+  private static final String LOCK_MSG = "lockMsg";
+  private static final String LOCK_SCOPE_PROPERTY = "property";
+  private static final String PATH_KEYS = "pathKeys";
+  private static final String PATH_KEY_DELIMITER = "\\|";
+
+  private enum LockScopeKey {
+    HelixLockScope,
+  }
+
+  /**
+   *  A class contains information to locate a lock znode
+   */
+  private class LockLocator {
+    private final String _zkAddress;
+    private final String _lockPath;
+
+    private LockLocator(String zkAddress, String lockPath) {
+      this._zkAddress = zkAddress;
+      this._lockPath = lockPath;
+    }
+
+    public String getZkAddress() {
+      return _zkAddress;
+    }
+
+    public String getLockPath() {
+      return _lockPath;
+    }
+  }
+
+  @ResponseMetered(name = HttpConstants.WRITE_REQUEST)
+  @Timed(name = HttpConstants.WRITE_REQUEST)
+  @POST
+  @Path("lock")
+  public Response tryLock(String content) {
+    System.out.println("Payload:\n" + content);
+    ZKDistributedNonblockingLock lock;
+    LockLocator lockLocator;
+    LockInfo lockInfo;
+    try {
+      // Parse payload data
+      ZNRecord lockContent = toZNRecord(content);
+
+      // Parse zk address and lock path
+      lockLocator = parseLockLocator(lockContent);
+
+      // Parse lock info such as user id, message, timeout
+      lockInfo = parseLockInfo(lockContent);
+
+      // Build a lock object
+      lock = new ZKDistributedNonblockingLock.Builder().setLockMsg(lockInfo.getMessage())
+          .setUserId(lockInfo.getOwner()).setTimeout(lockInfo.getTimeout())
+          .setZkAddress(lockLocator.getZkAddress()).setLockPath(lockLocator.getLockPath()).build();
```

Comment:
Let's be careful here - what would the zk address be set as if the user was using it on multi-zk mode?